WeakAuras/GenericTrigger.lua

```diff
@@ -1292,13 +1337,29 @@ function GenericTrigger.Add(data, region)
                     hasParam = true
                   end
                 elseif trueEvent:match("^UNIT_") then
+                  local includePets
+                  if i == "grouppets" then
+                    includePets = "PlayersAndPets"
+                    i = "group"
+                  elseif i == "grouppetsonly" then
+                    includePets = "PetsOnly"
+                    i = "group"
+                  end
                   MultiUnitLoop(
                     function(unit)
                       trigger_unit_events[unit] = trigger_unit_events[unit] or {}
                       tinsert(trigger_unit_events[unit], trueEvent)
                       isUnitEvent = true
-                    end, i
+                    end, i, includePets
                   )
```

Comment:
I have made several changes here.

a) The handling of multi-units is a bit confussing, and it turns out that GenericTrigger.Add should not expand it here. Instead it's expanded in LoadDisplays + LoadEvent do that. So the loop here was always unnecessary. (At least that's what I think)

b) I removed the special casing of UNIT_PET. Instead custom coders will need to add UNIT_PET:group to their event list, if they want to listen to that event. For internal triggers this works without UNIT_PET due to the UNIT_CHANGED internal events.

c) I added partypets, partypetsonly

d) I made the matching for grouppets etc, case insensitive, as that's what is done for matching group

e) I removed the warning that showed up for grouppets, grouppetsonly